### PR TITLE
py-pytecplot: new package

### PR DESCRIPTION
**Pull request by myan-rarcellino-nnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pytecplot/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-pytecplot/package.py]:
```suggestion
# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256.

Will defer merge decision until hear whether @adamjstewart wants to comment on this PR.

File: var/spack/repos/builtin/packages/py-pytecplot/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    and 3D volumetric visualization. Familiarity with Tecplot 360 and the
+    Tecplot 360 macro language is helpful, but not required."""
+
+    homepage = "https://www.tecplot.com/docs/pytecplot/"
+    pypi = "pytecplot/pytecplot-1.4.2.zip"
+
+    version("1.4.2", sha256="586a2ee947314ddd2f28be5523911dd298465b8f6a9145ba351866d5d695ef0d")
+
+    variant("numpy", default=False, description="Add numpy dependency.")
+    variant("ipython", default=False, description="Add ipython dependency.")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytecplot/package.py]: I usually try to keep the Spack variant names the same as pip. When you install with pip, you would use `pip install pytecplot[extras]` to get numpy/ipython. Should we do the same here, a single `+extras` variant?

Comment (myan-rarcellino-nnl) [reply var/spack/repos/builtin/packages/py-pytecplot/package.py]: yup. much more reasonable than 1 variant per extra dep

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**